### DISCUSSION TITLE
The handle method of Micro always generate Not-Found handler is not callable or is not defined in

### DISCUSSION THREAD

**Discussion by melatagia** [phalcon/cphalcon#15783] [Q&A]
Hi All,
When I call a route in the index file I always get the following message  **"PHP Fatal error:  Uncaught Phalcon\\Mvc\\Micro\\Exception: Not-Found handler is not callable or is not defined in..."**

This is the index.php
`use Phalcon\Mvc\Micro;
$app = new Micro();
$app->get(
    '/invoices/view/{id}',
    function ($id) {
        echo "<h1>Invoice #{$id}!</h1>";
    }
);
$app->handle(
    $_SERVER["REQUEST_URI"]
);`

This is the request curl -i -X GET   http://127.0.0.1/myapp/invoices/view/1

This is the configuration of the server 
- inux CentOS-79-64-minimal 3.10.0-1160.15.2.el7.x86_64
- PHP Version 8.0.12
- Phalcon 5.0.0alpha6

Can somebody know how to fix this ?
Thanks

**Comment by niden** [comment]
You need to add one more route that tells the router where a 404 would go. Have a look at this (method `notFound()`)

https://docs.phalcon.io/4.0/en/application-micro#not-found-404

**Reply by melatagia** [reply]
Thanks Nikolaos. I add it but when I call curl -i -X GET http://127.0.0.1/myapp/invoices/view/1 I get the 404 response, i.e. the notFound() is executed instead of the /invoices/view/ endpoint.

**Reply by melatagia** [reply]
Issue fixed. I used `$app->handle($_GET['_url']);` instead of ` $app->handle( $_SERVER["REQUEST_URI"] )`
Thanks
